pyccel/parser/semantic.py

```diff
@@ -2786,20 +2779,18 @@ def _visit_FunctionDef(self, expr, **settings):
 
             funcs = Interface(name, new_funcs)
             self.insert_function(funcs)
-
-        if vec_func:
-           self._visit_FunctionDef(vec_func, **settings)
-           vec_func = self.namespace.functions.pop(vec_name)
-           if isinstance(funcs, Interface):
-               funcs = list(funcs.funcs)+[vec_func]
-           else:
-               self.namespace.sons_scopes['sc_'+ name] = self.namespace.sons_scopes[name]
-               funcs = funcs.rename('sc_'+ name)
-               funcs = [funcs, vec_func]
-
-           funcs = Interface(name, funcs)
-           self.insert_function(funcs)
-
+#        TODO move this to codegen
```

Comment:
Can we actually move this to codegen within this pull request?

Reply:
I don't know if we should provide that for the moment as it was never working from the beginning, it was generating the right code but the behaviour in python and fortran was different

Reply:
> I don't know if we should provide that for the moment as it was never working from the beginning, it was generating the right code but the behaviour in python and fortran was different

I see. Then please create a dedicated "enhancement" issue and mention it here.